rg_instructor_analytics/models.py

```diff
@@ -0,0 +1,54 @@
+"""
+Models of the rg analytics.
+"""
+
+from django.contrib.auth.models import User
+from django.db.models import DateField, DateTimeField, ForeignKey, IntegerField, Model, PositiveSmallIntegerField
+
+from openedx.core.djangoapps.xmodule_django.models import CourseKeyField
+
+
+class EnrollmentTabCache(Model):
+    """
+    Cache for the Enrollment tab.
+    """
+
+    course_id = CourseKeyField(max_length=255, db_index=True)
+    created = DateField(auto_now_add=False, db_index=True)
+    unenroll = IntegerField(default=0)
+    enroll = IntegerField(default=0)
+    total = IntegerField(default=0)
+
+    class Meta:
+        """
+        Meta class.
+        """
+
+        unique_together = ('course_id', 'created',)
+
+
+class EnrollmentByStudent(Model):
+    """
+    Model for store last enrollment state of a user.
+    """
+
+    course_id = CourseKeyField(max_length=255, db_index=True)
+    student = ForeignKey(User, db_index=True)
+    last_update = DateTimeField(auto_now_add=False, db_index=True)
+    state = PositiveSmallIntegerField()
+
+    class Meta:
+        """
+        Meta class.
+        """
+
+        unique_together = ('course_id', 'student',)
+
+
+class TotalEnrollmentByCourse(Model):
```

Comment:
It seems you can avoid this model, and use the first one instead.

Reply:
done

Reply:
Or if you prefer to persist this data please use the cache.

Reply:
done.